### DISCUSSION TITLE
Getting app-check error when not using it.

### DISCUSSION THREAD

**Discussion by rolandszpond** [invertase/react-native-firebase#6177] [Q&A]
Hello, I'm getting the following error even though I haven't install this feature. I'm only using the core app package and the auth package.

You attempted to use "firebase.app('[DEFAULT]').appCheck" but this module could not be found. Ensure you have install and imported the '@react-native-firebase/app-check' package.

**Comment by rolandszpond** [comment]
Well, I found a fix for it... I think, the solution seems weird...

`const [ initializing, setInitializing ] = useState(true)`

    `useEffect(() => {
        const subscriber = auth().onAuthStateChanged((user) => {
            setUser(user)
            console.log(user)
            if (initializing) setInitializing(false)
        })
        return subscriber
    }, [])`

"if (initializing) setInitializing(false)" line is causing the error, which is just a state on my component, don't know why it's affecting the package at all? When I comment this line out, the error goes away and the app renders normally

**Comment by rolandszpond** [comment]
Seems like when trying to update any state when using `const [ value, setValue ] = useState()` inside of the `onAuthStateChanged` function is causing the error

**Reply by mikehardy** [reply]
Can you do me a favor and log this as an actual issue with versions specified, and an App.js that reproduces this? That is indeed pretty strange. You might see if you can reproduce it with the example app kicked out by https://github.com/mikehardy/rnfbdemo/blob/main/make-demo.sh especially if you comment out all the installs for modules you don't use (crashlytics, especially app-check, etc) to see if it shows up in a clean skeleton project
